gtsfm/bundle/rig_bundle_adjustment.py

```diff
@@ -0,0 +1,186 @@
+                BetweenFactorPose3(
+                    X(i),
+                    B(rig_idx),
+                    self._cam_T_imu[camera_type],
+                    gtsam.noiseModel.Isotropic.Sigma(CAM_POSE3_DOF, CAM_IMU_POSE_PRIOR_SIGMA),
+                )
+            )
+
+        # translate the relative pose priors between cams to IMUs, and add if not already present
+        imu_relative_pose_priors: Dict[Tuple[int, int], BetweenFactorPose3] = {}
+        for (i1, i2), i2Ti1_prior in relative_pose_priors.items():
+            if i2Ti1_prior is None or i1 not in cameras_to_model or i2 not in cameras_to_model:
+                continue
+
+            b1: int = self.__get_rig_idx(i1)
+            b2: int = self.__get_rig_idx(i2)
+
+            if b1 == b2:
+                # already captured in cam-IMU between factor.
+                continue
+
+            if (b1, b2) in imu_relative_pose_priors:
+                continue
+
+            if i2Ti1_prior.type == PosePriorType.HARD_CONSTRAINT:
+                raise ValueError("Encountered hard pose constraint in inter-rig camera pairs")
+
+            i2Ti1 = i2Ti1_prior.value
+            i1Tb1 = self._cam_T_imu[self.__get_camera_type(i1)]
+            i2Tb2 = self._cam_T_imu[self.__get_camera_type(i2)]
+
+            b2Tb1 = i2Tb2.inverse() * i2Ti1 * i1Tb1
+            imu_relative_pose_priors[(b1, b2)] = BetweenFactorPose3(
+                B(b2),
+                B(b1),
+                b2Tb1,
+                gtsam.noiseModel.Diagonal.Sigmas(i2Ti1_prior.covariance),
+            )
+
+        logger.info("Added %d between factors for IMUs", len(imu_relative_pose_priors))
+
+        for factor in imu_relative_pose_priors.values():
+            graph.push_back(factor)
+
+        return graph
+
+    def _pose_priors(
+        self,
+        absolute_pose_priors: List[Optional[PosePrior]],
+        initial_data: GtsfmData,
+        camera_for_origin: gtsfm_types.CAMERA_TYPE,
+    ) -> NonlinearFactorGraph:
+        """Add priors on the cam/IMU poses.
+
+        Right now, the function just adds a prior on 1 IMU frame to anchor the scene to origin.
+        """
+        graph = NonlinearFactorGraph()
+
+        # TODO(Ayush): start using absolute prior factors.
+        num_priors_added = 0
+
+        if num_priors_added == 0:
+            # Adding a prior to fix origin as no absolute prior exists.
+            rig_for_origin = self.__get_rig_idx(camera_for_origin)
+            wTi = initial_data.get_camera(camera_for_origin).pose()
+            wTb = wTi * self._cam_T_imu[self.__get_camera_type(camera_for_origin)]
+            graph.push_back(
+                PriorFactorPose3(
+                    B(rig_for_origin),
+                    wTb,
+                    gtsam.noiseModel.Isotropic.Sigma(CAM_POSE3_DOF, CAM_POSE3_PRIOR_NOISE_SIGMA),
+                )
+            )
+
+        return graph
+
+    def _initial_values(self, initial_data: GtsfmData) -> Values:
```

Comment:
Actually, I think we should get the initial values from the SLAM team and initialize the cameras from the IMU pose, right?

Reply:
This is resolved now as we don't have IMU pose anywhere. All camera poses will be available from averaging and we will use them.